crates/forge/bin/cmd/test/mod.rs

```diff
@@ -814,6 +827,49 @@ fn persist_run_failures(config: &Config, outcome: &TestOutcome) {
     }
 }
 
+/// Generate test report in JUnit XML report format.
+fn junit_xml_report(results: &BTreeMap<String, SuiteResult>, verbosity: u8) -> Report {
+    let mut junit_report = Report::new("Test run");
+    junit_report.set_timestamp(Utc::now());
+    for (suite_name, suite_result) in results {
+        let mut test_suite = TestSuite::new(suite_name);
+        test_suite.set_system_out(suite_result.summary());
+        for (test_name, test_result) in &suite_result.test_results {
+            let test_status = if test_result.status.is_failure() {
+                let mut status = TestCaseStatus::non_success(NonSuccessKind::Failure);
+                status.set_message(test_result.reason.clone().unwrap_or_default());
+                status
+            } else if test_result.status.is_skipped() {
+                TestCaseStatus::skipped()
+            } else {
+                TestCaseStatus::success()
+            };
+
+            let mut test_case = TestCase::new(test_name, test_status);
+            test_case.set_time(test_result.duration);
+
+            let mut sys_out =
+                format!("{} {} {}", test_result, test_name, test_result.kind.report());
+            if verbosity >= 2 && !test_result.logs.is_empty() {
+                sys_out.push_str("\\nLogs:\\n");
+                let console_logs = decode_console_logs(&test_result.logs);
+                for log in console_logs {
+                    sys_out.push_str(format!("  {log}\\n").as_str());
```

Comment:
use write! macro

Reply:
changed in https://github.com/foundry-rs/foundry/pull/8852/commits/143d0686f8a2efe4288db2575858d0f66e8afe3b